packages/components/src/text/text.styles.js

```diff
@@ -0,0 +1,14 @@
+/**
```

Comment:
This uses .styles files while the Card component uses a `styles` folder. Should we have a consistent convention?

Reply:
Yep! What convention would you like to use?

Reply:
I have a small preference for the `styles` folder maybe especially if this is something that ends up containing more that one style file (like the card does), but I don't have enough CSS in JS experience to make the call. Up to you folks :)

Reply:
Are there any existing conventions around nesting components? Are they nested in folders or kept at the same level as the parent?

e.g. 
‘’‘
toolbar/
  toolbar.js
  button/
    button.js
‘’’ 
or 
‘’‘
toolbar/
  toolbar.js
  button.js
‘’’

My vote would for a folder in the second case but not the first... but really it’s just a matter of preference and doesn’t matter whichever option is chosen. 

/cc @ItsJonQ

Reply:
in general when nesting components we do one of the following:

```
toolbar/
  index.js
  button/
    index.js
```

or 

```
toolbar/
  index.js
  button.js
```

Reply:
In general, this is something which might be less flexible that we might want to. See `Card` component and how links to subcomponents break in the block editor handbook:

https://developer.wordpress.org/block-editor/components/card/#sub-components

The current folder structure:

![Screen Shot 2019-11-18 at 13 04 33](https://user-images.githubusercontent.com/699132/69051224-04820c00-0a04-11ea-9b41-3495b3463c5f.png)

I did a quick test and the following would work out of the box with the current setup of the handbook integration:
![Screen Shot 2019-11-18 at 13 16 42](https://user-images.githubusercontent.com/699132/69051816-b241ea80-0a05-11ea-9f68-07e6953aa8f0.png)
Well, we would have to test. You could also move it to the root level and follow other components that have nested items:
- `Toolbar` & `ToolbarButton`
- `MenuGroup` & `MenuItem`

You can always tweak the setup if you think that the different approach would work better. However, this will require some additional effort to make sure it gets properly handled at https://developer.wordpress.org/block-editor/.

Reply:
I did a quick test with the following code changes:

```diff
diff --git a/docs/manifest-devhub.json b/docs/manifest-devhub.json
index 75573dd5e..6309394dc 100644
--- a/docs/manifest-devhub.json
+++ b/docs/manifest-devhub.json
@@ -605,6 +605,12 @@
 		"markdown_source": "../packages/components/src/button/README.md",
 		"parent": "components"
 	},
+	{
+		"title": "CardBody",
+		"slug": "card-body",
+		"markdown_source": "../packages/components/src/card/card-body/README.md",
+		"parent": "components"
+	},
 	{
 		"title": "Card",
 		"slug": "card",
diff --git a/packages/block-library/src/search/index.php b/packages/block-library/src/search/index.php
index 6cd909fc6..67bc90ff3 100644
--- a/packages/block-library/src/search/index.php
+++ b/packages/block-library/src/search/index.php
@@ -25,6 +25,12 @@ function render_block_core_search( $attributes ) {
 			$input_id,
 			$attributes['label']
 		);
+	} else {
+		$label_markup = sprintf(
+			'<label for="%s" class="wp-block-search__label screen-reader-text">%s</label>',
+			$input_id,
+			__( 'Search' )
+		);
 	}
 
 	$input_markup = sprintf(
@@ -45,7 +51,6 @@ function render_block_core_search( $attributes ) {
 	if ( isset( $attributes['className'] ) ) {
 		$class .= ' ' . $attributes['className'];
 	}
-
 	if ( isset( $attributes['align'] ) ) {
 		$class .= ' align' . $attributes['align'];
 	}
diff --git a/packages/components/src/card/README.md b/packages/components/src/card/README.md
index 4a401b021..9e27f52a2 100644
--- a/packages/components/src/card/README.md
+++ b/packages/components/src/card/README.md
@@ -26,7 +26,7 @@ Name | Type | Default | Description
 
 This component provides a collection of sub-component that can be used to compose various interfaces.
 
--   [`<CardBody />`](./docs/body.md)
+-   [`<CardBody />`](card-body/body.md)
 -   [`<CardDivider />`](./docs/divider.md)
 -   [`<CardFooter />`](./docs/footer.md)
 -   [`<CardHeader />`](./docs/header.md)
diff --git a/packages/components/src/card/card-body/index.js b/packages/components/src/card/card-body/index.js
index 9194d6011..b6a5d09a3 100644
--- a/packages/components/src/card/card-body/index.js
+++ b/packages/components/src/card/card-body/index.js
@@ -6,8 +6,8 @@ import classnames from 'classnames';
 /**
  * Internal dependencies
  */
-import { BodyUI } from './styles/card-styles';
-import { useCardContext } from './context';
+import { BodyUI } from '../styles/card-styles';
+import { useCardContext } from '../context';
 
 export const defaultProps = {
 	isShady: false,
@@ -20,7 +20,7 @@ export function CardBody( props ) {
 	const { size } = mergedProps;
 
 	const classes = classnames(
-		'components-card__body',
+		'components-card-body',
 		isShady && 'is-shady',
 		size && `is-size-${ size }`,
 		className
diff --git a/packages/components/src/card/card-body/stories/index.js b/packages/components/src/card/card-body/stories/index.js
index a1dabe5bf..462b0e035 100644
--- a/packages/components/src/card/card-body/stories/index.js
+++ b/packages/components/src/card/card-body/stories/index.js
@@ -8,9 +8,9 @@ import { boolean, text } from '@storybook/addon-knobs';
 /**
  * Internal dependencies
  */
-import Card from '../index';
-import CardBody from '../card-body/body';
-import { getCardStoryProps } from './_utils';
+import Card from '../../';
+import CardBody from '../';
+import { getCardStoryProps } from '../../stories/_utils';
 
 export default { title: 'Components|Card/Body', component: CardBody };
 
diff --git a/packages/components/src/card/stories/_index.js b/packages/components/src/card/stories/_index.js
index 590a988df..763214ce2 100644
--- a/packages/components/src/card/stories/_index.js
+++ b/packages/components/src/card/stories/_index.js
@@ -10,7 +10,7 @@ import { text, boolean } from '@storybook/addon-knobs';
  * Internal dependencies
  */
 import Card from '../index';
-import CardBody from '../body';
+import CardBody from '../card-body';
 import CardFooter from '../footer';
 import CardHeader from '../header';
 import { getCardStoryProps } from './_utils';
diff --git a/packages/components/src/card/stories/divider.js b/packages/components/src/card/stories/divider.js
index c506cef28..8f9c8f1f9 100644
--- a/packages/components/src/card/stories/divider.js
+++ b/packages/components/src/card/stories/divider.js
@@ -2,7 +2,7 @@
  * Internal dependencies
  */
 import Card from '../index';
-import CardBody from '../body';
+import CardBody from '../card-body';
 import CardDivider from '../divider';
 import { getCardStoryProps } from './_utils';
 
diff --git a/packages/components/src/card/stories/media.js b/packages/components/src/card/stories/media.js
index 4ecade543..dec853175 100644
--- a/packages/components/src/card/stories/media.js
+++ b/packages/components/src/card/stories/media.js
@@ -10,7 +10,7 @@ import styled from '@emotion/styled';
  * Internal dependencies
  */
 import Card from '../index';
-import CardBody from '../body';
+import CardBody from '../card-body';
 import CardFooter from '../footer';
 import CardHeader from '../header';
 import CardMedia from '../media';
diff --git a/packages/components/src/card/test/body.js b/packages/components/src/card/test/body.js
index dbbe21cdd..1647472e2 100644
--- a/packages/components/src/card/test/body.js
+++ b/packages/components/src/card/test/body.js
@@ -6,13 +6,13 @@ import { shallow } from 'enzyme';
 /**
  * Internal dependencies
  */
-import CardBody from '../body';
+import CardBody from '../card-body';
 
 describe( 'CardBody', () => {
 	describe( 'basic rendering', () => {
 		test( 'should have components-card className', () => {
 			const wrapper = shallow( <CardBody /> );
-			const cardBody = wrapper.find( '.components-card__body' );
+			const cardBody = wrapper.find( '.components-card-body' );
 
 			expect( cardBody.length ).toBe( 1 );
 		} );
@@ -33,14 +33,14 @@ describe( 'CardBody', () => {
 	describe( 'modifiers', () => {
 		test( 'should be able to render size modifier', () => {
 			const wrapper = shallow( <CardBody size="large" /> );
-			const cardBody = wrapper.find( '.components-card__body' );
+			const cardBody = wrapper.find( '.components-card-body' );
 
 			expect( cardBody.hasClass( 'is-size-large' ) ).toBe( true );
 		} );
 
 		test( 'should be able to render shady modifier', () => {
 			const wrapper = shallow( <CardBody isShady /> );
-			const cardBody = wrapper.find( '.components-card__body' );
+			const cardBody = wrapper.find( '.components-card-body' );
 
 			expect( cardBody.hasClass( 'is-shady' ) ).toBe( true );
 		} );
diff --git a/packages/components/src/card/test/index.js b/packages/components/src/card/test/index.js
index 16b4d3158..2eee9b11b 100644
--- a/packages/components/src/card/test/index.js
+++ b/packages/components/src/card/test/index.js
@@ -7,7 +7,7 @@ import { mount, shallow } from 'enzyme';
  * Internal dependencies
  */
 import Card from '../';
-import CardBody from '../body';
+import CardBody from '../card-body';
 import CardDivider from '../divider';
 import CardFooter from '../footer';
 import CardHeader from '../header';
@@ -70,7 +70,7 @@ describe( 'Card', () => {
 					<CardBody>Hello</CardBody>
 				</Card>
 			);
-			const cardBody = card.find( '.components-card__body' ).first();
+			const cardBody = card.find( '.components-card-body' ).first();
 
 			expect( cardBody.hasClass( 'is-size-extraSmall' ) ).toBe( true );
 		} );
@@ -81,7 +81,7 @@ describe( 'Card', () => {
 					<CardBody size="large">Hello</CardBody>
 				</Card>
 			);
-			const cardBody = card.find( '.components-card__body' ).first();
+			const cardBody = card.find( '.components-card-body' ).first();
 
 			expect( cardBody.hasClass( 'is-size-large' ) ).toBe( true );
 		} );
diff --git a/packages/components/src/index.js b/packages/components/src/index.js
index 3803719c2..62fcfc55d 100644
--- a/packages/components/src/index.js
+++ b/packages/components/src/index.js
@@ -6,7 +6,7 @@ export { default as BaseControl } from './base-control';
 export { default as Button } from './button';
 export { default as ButtonGroup } from './button-group';
 export { default as Card } from './card';
-export { default as CardBody } from './card/body';
+export { default as CardBody } from './card/card-body';
 export { default as CardDivider } from './card/divider';
 export { default as CardFooter } from './card/footer';
 export { default as CardHeader } from './card/header';
```

Reply:
Given that information I think there'll _usually_ be a single style file in each component or sub-component folder so I'd personally lean towards `<component>.styles.js` and `<component.styles.tests.js` where necessary. However it looks like there's already a pattern of using folders e.g. `tests` and `stories` even though they often contain a single file.

In the case of this simple component, the exports are styled components so adhering to the `styles` folder pattern, should all of the files be nested in the `styles` folder? Where should the tests go? Should they be nested in `styles` too?

Is this how you'd like to see it structured?

```
text/
  text.js // doesn't exist in this case
  styles/
    tests/
      text.js
      text.native.js
    mixins.js
    text.js
    text.native.js
```

This is turning into a lengthy discussion, should we move it to a seperate issue and we can come back and refactor whatever components we have when a decision is made?

Reply:
I've created a [seperate issue](https://github.com/WordPress/gutenberg/issues/18677) to continue this discussion. When a decision is made, I'll come back and refactor any CSS-in-JS components that have made it into the repository before that decision is made.